### PR TITLE
[Test] Refactor common utilities function for e2e tests

### PR DESCRIPTION
**Pull request by fra98**
# Description

This PR refactors common utilities function for e2e tests

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -253,6 +254,7 @@ e2e: $(E2E_TARGETS)
 
 e2e-dir:
 	mkdir -p "${BINDIR}"
+	export PATH="${BINDIR}:${PATH}"
```
Comment (aleoli) [inline Makefile:257]: Are you sure that this export will also be available for the next make steps?

**Review by aleoli — APPROVED** [review]

**Review by aleoli — APPROVED** [review]

File: test/e2e/pipeline/utils.sh [hunk 2]
```diff
@@ -1,6 +1,10 @@
 #!/usr/bin/env bash
 #shellcheck disable=SC1091
 
+set -e           # Fail in case of error
+set -o nounset   # Fail if undefined variables are used
+set -o pipefail  # Fail if one of the piped commands fails
```
Comment (cheina97) [inline test/e2e/pipeline/utils.sh:6]: Are you sure it is necessary here? I would suggest to add error handling only in files that import utilities.

File: test/e2e/pipeline/infra/k3s/pre-requirements.sh [hunk 3]
```diff
@@ -27,41 +29,24 @@ error() {
...
-    armv6*) ARCH="armv6";;
-    armv7*) ARCH="arm";;
-    aarch64) ARCH="arm64";;
-    x86) ARCH="386";;
-    x86_64) ARCH="amd64";;
-    i686) ARCH="386";;
-    i386) ARCH="386";;
-    *) echo "Error architecture '${ARCH}' unknown"; exit 1 ;;
-  esac
+source "$WORKDIR/../../utils.sh"
```
Comment (cheina97) [inline test/e2e/pipeline/infra/k3s/pre-requirements.sh:35]: Can you add shellcheck directive for completion?

File: test/e2e/pipeline/infra/kind/pre-requirements.sh [hunk 4]
```diff
@@ -27,41 +29,23 @@ error() {
...
-  esac
+FILEPATH=$(realpath "$0")
+WORKDIR=$(dirname "$FILEPATH")
 
-  OS=$(uname |tr '[:upper:]' '[:lower:]')
-  case "$OS" in
-    # Minimalist GNU for Windows
-    "mingw"*) OS='windows'; return ;;
-  esac
+source "$WORKDIR/../../utils.sh"
```
Comment (cheina97) [inline test/e2e/pipeline/infra/kind/pre-requirements.sh:35]: Can you add shellcheck directive for completion?

**Review by cheina97 — APPROVED** [review]

File: test/e2e/pipeline/utils.sh [hunk 5]
```diff
@@ -123,4 +119,21 @@ function install_helm {
...
+
+function install_metrics_server() {
+    local kubeconfig=$1
+
+    "${HELM}" repo add metrics-server https://kubernetes-sigs.github.io/metrics-server/
+    "${HELM}" upgrade --install metrics-server metrics-server/metrics-server \
+        --set 'args={"--kubelet-insecure-tls=true"}' \
+        --namespace kube-system --kubeconfig "${kubeconfig}"
+    "${KUBECTL}" -n kube-system rollout status deployment metrics-server --kubeconfig "${kubeconfig}"
 }
```
Comment (cheina97) [inline test/e2e/pipeline/utils.sh:139]: Please, add a line at the end of the file

**Review by cheina97 — APPROVED** [review]

**Pull request merged**